blackbox/docs/administration/ssl.txt

```diff
@@ -0,0 +1,99 @@
+using ``ssl.keystore_filepath`` setting.
+
+.. NOTE::
+
+    Make sure that the keystore file has the right permissiong and is accessible by system user ``crate``.
+
+Also define the password needed to decrypt the keystore by using the ``ssl.keystore_password`` setting.
+
+Optionally if a separate key password is defined when creating the keystore define it by using
+the ``ssl.keystore_key_password`` setting.
+
+A full list of the settings needed to configure SSL/TLS can be found :ref:`here <ssl_config>`.
+
+Configure a separate TrustStore
+-------------------------------
+
+Trusted CA certificates can also be stored in a node's keystore but optionally a separate truststore can be
+used to store them.
+
+
+If you want to use a separate truststore create a node truststore and import the CA certificate(s) you want
+to trust. Once the truststore is prepared define the absolute file path of the truststore ``.jks`` file on the node
+using the ``ssl.truststore_filepath`` setting.
+
+.. NOTE::
+
+    Make sure that the truststore file has the right permissiong and is accessible by system user ``crate``.
+
+Also define the password needed to decrypt the keystore by using the ``ssl.truststore_password`` setting.
+
+A full list of the settings needed to configure SSL/TLS can be found :ref:`here <ssl_config>`.
+
+
+Connecting to a CrateDB node using Postgres Wire Protocol through secured communications
+----------------------------------------------------------------------------------------
+
+Connect to a CrateDB node using JDBC
+....................................
+
+JDBC needs to validate the CrateDB node's identity by checking that the node certificate is signed by a trusted
+authority. If the certificate is signed by a global certificate authority (CA), there is nothing further to do because
+Java comes with copies of the most common CA's certificates. If you have a self-signed certificate though,
+you need to make this available to the Java client to enable it to validate the server's certificate.
+
+For that you need to configure a truststore which contains the node's certificate and start provide the path to
+the trustore file along with the password to your java application::
+
+    java -Djavax.net.ssl.trustStore=mystore -Djavax.net.ssl.trustStorePassword=mypassword com.mycompany.MyApp
+
+In case you face any issues extra debugging information is available by adding ``-Djavax.net.debug=ssl``
+to your command line.
+
+Last but not least the connection parameter ``ssl=true`` must be added to the connection URL so that the JDBC driver
+will try and establish an SSL connection.
+
+Connect to a CrateDB node using psql
+....................................
+
+In order to connect to a CrateDB node using SSL/TLS with the psql client simply define ``sslmode=verify-full`` in your
+connection string. E.g.::
+
+    psql "sslmode=verify-full host=localhost port=5432"
```

Comment:
The default is `sslmode=prefer` which will use ssl if possible but not verify the cert chain: https://www.postgresql.org/docs/9.1/static/libpq-ssl.html#LIBPQ-SSL-SSLMODE-STATEMENTS